plugins/poll/spec/models/poll_spec.rb

```diff
@@ -81,6 +81,11 @@
 
       expect(poll.post).to eq(post)
     end
+  end
 
+  it "is not throwing an error when double save" do
+    post = Fabricate(:post, raw: "[poll]\n- A\n- B\n[/poll]")
+    post.custom_fields["expert"] = true
```

Comment:
Not sure how adding this line test for "double save" though. I _think_ it requires the `discourse-category-experts` plugin to be installed?

Reply:
You are right, it is not neccessary.

Without fix, just save! twice is raising an error:
```
 post = Fabricate(:post, raw: "[poll]\n- A\n- B\n[/poll]")
 post.save!
```

The reason for that is that raw is same and extraced_polls were not reset:
https://github.com/discourse/discourse/blob/master/plugins/poll/plugin.rb#L489